tests/test_parseutils.py

```diff
@@ -3,52 +3,114 @@
+
+def test_simple_select_single_table_schema_qualified():
+    tables = extract_tables('select * from abc.def')
+    assert tables.to_dict('list') == \
+           {'schema': ['abc'], 'table': ['def'], 'alias': [None]}
 
 def test_simple_select_multiple_tables():
     tables = extract_tables('select * from abc, def')
-    assert tables == ['abc', 'def']
+    assert tables.to_dict('list') == \
+           {'schema': [None, None],
+            'table': ['abc', 'def'],
+            'alias': [None, None]}
+
+def test_simple_select_multiple_tables_schema_qualified():
+    tables = extract_tables('select * from abc.def, ghi.jkl')
+    assert tables.to_dict('list') == \
+           {'schema': ['abc', 'ghi'],
+            'table': ['def', 'jkl'],
+            'alias': [None, None]}
 
 def test_simple_select_with_cols_single_table():
     tables = extract_tables('select a,b from abc')
-    assert tables == ['abc']
+    assert tables.to_dict('list') == \
+           {'schema': [None], 'table': ['abc'], 'alias': [None]}
+
+def test_simple_select_with_cols_single_table_schema_qualified():
+    tables = extract_tables('select a,b from abc.def')
+    assert tables.to_dict('list') == \
+           {'schema': ['abc'], 'table': ['def'], 'alias': [None]}
 
 def test_simple_select_with_cols_multiple_tables():
     tables = extract_tables('select a,b from abc, def')
-    assert tables == ['abc', 'def']
+    assert tables.to_dict('list') == \
+           {'schema': [None, None],
+            'table': ['abc', 'def'],
+            'alias': [None, None]}
+
+def test_simple_select_with_cols_multiple_tables():
+    tables = extract_tables('select a,b from abc.def, def.ghi')
+    assert tables.to_dict('list') == \
+           {'schema': ['abc', 'def'],
+            'table': ['def', 'ghi'],
+            'alias': [None, None]}
 
 def test_select_with_hanging_comma_single_table():
     tables = extract_tables('select a, from abc')
-    assert tables == ['abc']
+    assert tables.to_dict('list') == \
+           {'schema': [None],
+            'table': ['abc'],
+            'alias': [None]}
 
 def test_select_with_hanging_comma_multiple_tables():
     tables = extract_tables('select a, from abc, def')
-    assert tables == ['abc', 'def']
+    assert tables.to_dict('list') == \
+           {'schema': [None, None],
+            'table': ['abc', 'def'],
+            'alias': [None, None]}
+
+def test_select_with_hanging_period_multiple_tables():
+    tables = extract_tables('SELECT t1. FROM tabl1 t1, tabl2 t2')
+    assert tables.to_dict('list') == \
+           {'schema': [None, None],
+            'table': ['tabl1', 'tabl2'],
+            'alias': ['t1', 't2']}
 
 def test_simple_insert_single_table():
     tables = extract_tables('insert into abc (id, name) values (1, "def")')
-    assert tables == ['abc']
+    assert tables.to_dict('list') == \
+           {'schema': [None], 'table': ['abc'], 'alias': ['abc']}
+
+@pytest.mark.xfail
```

Comment:
oh that makes sense. Pycharm was apparently just autoimporting pytest for me. I'll fix it in a minute